### ISSUE TITLE
stringify sparse arrays

### ISSUE THREAD

**Issue by twada** [twada/stringifier#4]
Should stringify sparse arrays correctly

actual:
```js
var input = [];
input[1] = 'unicorn';
console.log(stringify(input));  // => ["unicorn"]
```

expected:
```js
var input = [];
input[1] = 'unicorn';
console.log(stringify(input));  // => [,"unicorn"]
```

See: [t.same() and sparse arrays · Issue #104 · sindresorhus/ava](https://github.com/sindresorhus/ava/issues/104)

**Comment by schnittstabil** [comment]
@twada please note:

```js
var sparse = [];
sparse[1] = 'unicorn';
console.log(sparse);      // => [ , 'unicorn' ]
console.log(0 in sparse); // => false
```
But:
```js
var input = [undefined, 'unicorn'];
console.log(input);      // =>[ undefined, 'unicorn' ]
console.log(0 in input); // => true
```

**Comment by twada** [comment]
@schnittstabil Thanks, that's the point.

**Comment by twada** [comment]
@schnittstabil Just released [stringifier 1.2.1](https://github.com/twada/stringifier/blob/master/CHANGELOG.md#121-2015-10-26). Thank you for reporting!
